### DISCUSSION TITLE
Quartz Clustering Support with Separate Scheduling and Triggering Components

### DISCUSSION THREAD

**Discussion by sharon-birnboim-glbe** [quartznet/quartznet#2479] [Q&A] [answered]
Hi,

I have two services using Quartz with JobStore (PostgreSQL):

* One service is responsible for scheduling jobs and its scheduler is set to standby mode (scheduler.Standby()).
* The other service acts as a worker, triggering the jobs based on the Quartz tables populated by the first service.

These services will be deployed as separate units, with the first service having 2 nodes (replicas) and the second service also having 2 nodes.

What is the best way to configure the scheduler on each of these services for clustering?
 If they all use the same scheduler name, will they function correctly in a cluster?

**Comment by sharon-birnboim-glbe** [comment, marked as answer]
The scheduler can be configured as usual with a unique instance id per each service (node)

**Comment by lahma** [comment]
> One service is responsible for scheduling jobs and its scheduler is set to standby mode (scheduler.Standby()).

The idiomatic way is to not start the scheduler and ensure that it's using `ZeroSizeThreadPool`. This way the controlling node can never take work into execution.
